x-pack/metricbeat/module/azure/azure.go

```diff
@@ -24,16 +25,50 @@ func newModule(base mb.BaseModule) (mb.Module, error) {
 	return &base, nil
 }
 
+// MetricStore holds the accumulated metric definitions with a mutex for synchronization.
+type MetricStore struct {
+	sync.Mutex
+	accumulatedMetrics []Metric
+}
+
+func (s *MetricStore) AddMetric(metric Metric) {
+	s.Lock()         // Acquire the lock
+	defer s.Unlock() // Ensure the lock is released when the function returns
+	s.accumulatedMetrics = append(s.accumulatedMetrics, metric)
+}
+
+func (s *MetricStore) GetMetrics() []Metric {
+	s.Lock()         // Acquire the lock to prevent writes while reading
+	defer s.Unlock() // Ensure the lock is released when the function returns
+	return s.accumulatedMetrics
+}
+
+// ClearMetrics clears all accumulated metrics
+func (s *MetricStore) ClearMetrics() {
+	s.Lock()                          // Acquire the lock
+	defer s.Unlock()                  // Ensure the lock is released when the function returns
+	s.accumulatedMetrics = []Metric{} // Reset the accumulated metrics slice
+}
+
+// Size returns the size of the store
+func (s *MetricStore) Size() int {
+	return len(s.GetMetrics())
+}
+
 // MetricSet holds any configuration or state information. It must implement
 // the mb.MetricSet interface. And this is best achieved by embedding
 // mb.BaseMetricSet because it implements all of the required mb.MetricSet
 // interface methods except for Fetch.
 type MetricSet struct {
 	mb.BaseMetricSet
-	Client     *Client
-	MapMetrics mapResourceMetrics
+	Client         *Client
+	MapMetrics     mapResourceMetrics
+	BatchClient    *BatchClient
+	ConcMapMetrics concurrentMapResourceMetrics // In combination with BatchClient only
 }
 
+var monitorMetricsets = []string{"monitor", "container_registry", "container_instance", "container_service", "compute_vm", "compute_vm_scaleset", "database_account"}
```

Comment:
We are not including `storage` because we it isn't working with the Batch API yet, right?

Reply:
I decided to not include it for now. First of all it is not `monitor` metricset but the main reason is that while testing the documents returned with the batch API are of slightly different value to the standard api.
It is related to the documents dimensions but it will require extra investigation. Can be included later

Reply:
Correctness first.